livekit-rtc/livekit/rtc/video_stream.py

```diff
@@ -37,34 +41,81 @@ class VideoStream:
     def __init__(
         self,
         track: Track,
-        *,
         loop: Optional[asyncio.AbstractEventLoop] = None,
         capacity: int = 0,
         format: Optional[proto_video_frame.VideoBufferType.ValueType] = None,
+        **kwargs,
```

Comment:
I think it's OK to directly put participant here too?
VideoStream.from_participant and AudioStream.from_participant are not necessary I think. We can use typing.overload on the constructor?

Reply:
I prefer from_participant because it's the only one that uses track_source as well. So track_source and participant are used for the participant path and track is used for the other path. This is also why I chose to use kwargs (so it doesn't show up as an option during autocomplete) although I'll ready more about typing.overload to see if that could be useful here

Reply:
overloading the types is kind of clunky here because the args are positional and there are default values. So in the second overload I would need to enforce kwargs with `*` while keeping the original overload in tact. Do you still think it's better to remove `from_participant` in favor of `__init__` overloads? My personal preference is still the from_participant

Reply:
offline discussion. Keep as-is + introduce `from_track` for symmetry